framework/SupervisedLearning/ScikitLearn/ScikitLearnBase.py

```diff
@@ -74,17 +75,41 @@ def initializeModel(self, settings):
       @ In, settings, dict, the dictionary containin the parameters/settings to instanciate the model
       @ Out, None
     """
-    settings = self.updateSettings(settings)
-    self.settings = settings
-    self.model.set_params(**settings)
+    if self.settings is None:
+      self.settings = settings
```

Comment:
should we set self.settings = settings only when settings is None?

Reply:
I think so, self.settings is the initial settings for the model, which is only initialized when we parse the input XML. It is can be used to reset the model. That's why I added a 'if' check here.